channeldb/kvdb/etcd/embed.go

```diff
@@ -0,0 +1,73 @@
+package etcd
+
+import (
+	"fmt"
+	"net"
+	"net/url"
+	"time"
+
+	"github.com/coreos/etcd/embed"
+)
+
+const (
+	// readyTimeout is the time until the embedded etcd instance
+	// should start.
+	readyTimeout = 10 * time.Second
+)
+
+// getFreePort returns a random open TCP port.
+func getFreePort() int {
+	ln, err := net.Listen("tcp", "[::]:0")
+	if err != nil {
+		panic(err)
+	}
+
+	port := ln.Addr().(*net.TCPAddr).Port
+
+	err = ln.Close()
+	if err != nil {
+		panic(err)
+	}
+
+	return port
+}
+
+// NewEmbeddedEtcdInstance creates an embedded etcd instance for testing,
+// listening on random open ports. Returns the connection config and
+// a cleanup func that will stop the etcd instance.
+func NewEmbeddedEtcdInstance(path string) (*ConnConfig, func(), error) {
+	cfg := embed.NewConfig()
+	cfg.Dir = path
+
+	// To ensure that we can submit large transactions.
+	cfg.MaxTxnOps = 1000
```

Comment:
Is this tuned based on the current testnet testing? In the end though, I think this is only meant to be used in tests, correct?

Reply:
This is only used in test and required to make channeldb tests succeed. You can test it yourself with: `LND_TEST_DB_BACKEND=etcd go test -v` in the channeldb package.